client/src/App.tsx

```diff
@@ -8,8 +8,20 @@ import { ROUTES } from './config/routes';
 import Settings from '@/pages/Settings';
 import BaseLayout from '@/pages/BaseLayout';
 import OAuth from '@/pages/Oauth';
+import { useLocales } from '@/config/i18n';
+import { usePreferences } from '@/context/PreferencesContext';
+import { useEffect } from 'react';
 
 function App() {
+  const { preferences } = usePreferences();
+  const { changeLanguage, currentLanguage } = useLocales();
+
+  useEffect(() => {
```

Comment:
do this in PreferencesContext in the useeffect hook. A similar thing is done there